### PR TITLE
Add commitCount to all updateJob and createJob data

### PR DESCRIPTION
**Pull request by iwiznia**
Needed to fix this specific instance  https://github.com/Expensify/Expensify/issues/88302#issuecomment-429371913

Hold on https://github.com/Expensify/Bedrock/pull/514

Tests:
- Create a job without having done any bedrock call in any cluster. Check it created a job without adding any commitCounts.
- Create a job having done a call in only one cluster. Check it created the job and commitCounts contains only the count for that one cluster.
- Create a job having done a call in more than one cluster. Check it created the job and commitCounts contains all the counts for the clusters.
- Run BWM on the jobs created above and check that they correctly pass the commitCount (note: the webrock commitCount is updated again [here](https://github.com/Expensify/Bedrock-PHP/blob/1236fe20c4c3b82cbb50a56d67d918733edd765f/bin/BedrockWorkerManager.php#L287), which is correct given that we want the commitCount of the GetJobs to be included, which will always be greater than the commitCount for when the job was queued).

### CODE REVIEW

**Comment by iwiznia** [comment]
I need to do a couple more tests for this...

**Comment by iwiznia** [comment]
ok, this is ready.

**Comment by iwiznia** [comment]
Will create the web part when this is merged.

File: src/Client.php [hunk 1]
```diff
@@ -375,6 +398,7 @@ public function call($method, $headers = [], $body = '')
             } catch (BedrockError $e) {
                 // This error happen after sending some data to the server, so we only can retry it if it is an idempotent command
                 $this->markHostAsFailed($hostName);
+                /* @phan-suppress-next-line PhanTypeInvalidDimOffset for some reason phan says idempotent does not exist, but I have the ?? so it should not matter */
```
Comment (fnwbr) [inline src/Client.php]: Did you add this after you updated Phan?

File: src/Jobs.php [hunk 2]
```diff
@@ -197,6 +198,10 @@ public function createJobs(array $jobs): array
                 unset($jobs[$i]['priority']);
             }
         }
+        $commitCounts = Client::getCommitCounts();
+        foreach ($jobs as $i => $job) {
```
Comment (fnwbr) [inline src/Jobs.php]: NAB: I personally prefer `array_walk` with passing the first param as a reference over the `$jobs[$i]` stuff.

```php
array_walk($jobs, function (&$job) use ($commitCounts) {
    $job['data'] = array_merge($job['data'] ?? [], count($commitCounts) ? ['commitCounts' => $commitCounts] : []);
}
```

File: src/Client.php (hunk 1 above)
Comment (iwiznia) [reply src/Client.php]: Yes, phan was broken and not running (not sure since when)

**Comment by pecanoro** [comment]
Without waiting for my review... 😅

**Comment by fnwbr** [comment]
Ionatan asked me to 😬

File: src/Jobs.php (hunk 2 above)
Comment (iwiznia) [reply src/Jobs.php]: Yeah, I'll take the NAB, I don't like that with that awful reference there and everything.

Comment (pecanoro) [reply src/Jobs.php]: array_walk is way more difficult to read/understand

Comment (iwiznia) [reply src/Jobs.php]: lol 
image

Comment (pecanoro) [reply src/Jobs.php]: Maybe if you try with `array_walk` instead of `array_wakk` you will something 😄

Comment (iwiznia) [reply src/Jobs.php]: You might think so...
image

**Pull request merged**